openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMModelCreator.java

```diff
@@ -274,6 +277,23 @@ public LightGBMBinaryClassificationModel loadModel(final Path modelPath,
             errorsBuilder.add(new ParamValidationError(ERROR_MSG_NON_BINARY_TARGET));
         }
 
+        if (!Files.exists(modelPath)) {
+            logger.error(ERROR_MSG_PREFIX_CANNOT_FIND_MODEL_FILE + " in filesystem ({}).", modelPath);
+            errorsBuilder.add(new ParamValidationError(ERROR_MSG_PREFIX_CANNOT_FIND_MODEL_FILE + " in filesystem."));
+
+            return errorsBuilder.build();
+        }
+
+        if (Files.isDirectory(modelPath) && !Files.exists(modelPath.resolve(MODEL_BINARY_RESOURCE_FILE_NAME))) {
+            logger.error(
+                    "Error loading model from directory ({}). File {} not found.",
+                    modelPath, MODEL_BINARY_RESOURCE_FILE_NAME
+            );
+            errorsBuilder.add(new ParamValidationError(String.format("%s %s inside folder.",
+                    ERROR_MSG_PREFIX_CANNOT_FIND_MODEL_FILE, MODEL_BINARY_RESOURCE_FILE_NAME)
+            ));
+        }
```

Comment:
Why remove the else if and add an extra return?

It highlighted the two mutually exclusive alternatives: the model is either in single-file form or folder-form with a specific structure.